src/core/client_channel/subchannel.cc

```diff
@@ -543,6 +550,50 @@ class Subchannel::ConnectionStateWatcher final
     RefCountedPtr<ConnectedSubchannel> connected_subchannel =
         std::move(subchannel_->connected_subchannel_);
     if (connected_subchannel == nullptr) return;
+    // Record disconnection metrics
+    if (subchannel_->stats_plugin_group_ != nullptr) {
+      // Determine disconnect reason from DisconnectInfo
+      std::string disconnect_reason;
+      if (subchannel_->shutdown_) {
+        disconnect_reason = "subchannel shutdown";
+      } else {
+        switch (disconnect_info.reason) {
+          case DisconnectReason::kGoaway:
+            if (disconnect_info.http2_error_code.has_value()) {
+              disconnect_reason = absl::StrCat(
+                  "goaway_",
```

Comment:
All of the underscores in these label values should be spaces instead, as per the gRFC.